modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java

```diff
@@ -177,6 +183,139 @@ public void testAddObjectValidationRule() throws Exception {
 						RandomTestUtil.randomString()
 					).build())));
 
+		ObjectField textObjectField = _objectFieldLocalService.fetchObjectField(
+			_objectDefinition.getObjectDefinitionId(), "textObjectField");
+
+		AssertUtils.assertFailure(
+			ObjectValidationRuleSettingValueException.
+				CompositeKeyMustHaveMaxObjectFields.class,
+			"Add a maximum of five object fields to create unique composite " +
+				"keys",
+			() -> _addObjectValidationRule(
+				ObjectValidationRuleConstants.ENGINE_TYPE_COMPOSITE_KEY,
+				errorLabelMap, StringPool.BLANK, nameLabelMap,
+				ObjectValidationRuleConstants.OUTPUT_TYPE_FULL_VALIDATION,
+				StringPool.BLANK, false,
+				Arrays.asList(
```

Comment:
Instead of pass a list that way, you can create a list before the _assertFailure_, filling it using a loop and then pass it inside __addObjectValidationRule_

With that, I believe the method **_addCustomObjectField** can be inline inside the loop and [this one](https://github.com/liferay-objects/liferay-portal/pull/2871/files#diff-08df493d2d5ae4e11e74c839905b8135e4af78aac23eb4ad66067d06374ce2f3R33-R48) will become not needed anymore